scripts/test.sh

```diff
@@ -97,33 +99,31 @@ for i in "${ignored_packages[@]}"; do
   serial_packages=(${serial_packages[@]//*$i*})
 done
 
-if [ "${1:-""}" = "" ]; then
+if [[ -z "${restrict_testing_to}" ]]; then
   for dir in "${packages[@]}"; do
     pushd "$dir"
       ginkgo -p --race -randomizeAllSpecs -randomizeSuites \
-        -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-        ${@:2}
+        -ldflags="-extldflags=-Wl,--allow-multiple-definition"
     popd
   done
   for dir in "${serial_packages[@]}"; do
     pushd "$dir"
       ginkgo --nodes "${SERIAL_NODES}" --race -randomizeAllSpecs -randomizeSuites -failFast \
-        -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-        ${@:2}
+        -ldflags="-extldflags=-Wl,--allow-multiple-definition"
     popd
   done
 else
-  dir="${@: -1}"
+  dir="${restrict_testing_to}"
   dir="${dir#./}"
   for package in "${serial_packages[@]}"; do
     if [[ "${dir##$package}" != "${dir}" ]]; then
-      ginkgo --race -randomizeAllSpecs -randomizeSuites -failFast \
-        -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-        "${@}"
+      pushd "${dir}"
```

Comment:
[Medium] I think that you should be making the `pushd` to `package`, i.e. `pushd package` because `package` is the specific directory for the test suite. The `dir` variable from above is a parent directory to which the testing should be restricted to but it might contain several other testing suites.

Reply:
The if statement here is really confusing. As far as i can tell it's effectively doing `if [[ "${dir}" == "${package}" ]]; then`.. The dir variable passed in above is the directly we explicitly want to test, and we don't recurse to sub-packages in ginkgo

Reply:
Well now it makes sense. Then that loop was the confusing bit that had probably been leftover from before.